contracts/contracts/OpenAiChatGpt.sol

```diff
@@ -126,15 +129,17 @@ contract OpenAiChatGpt {
             run.messages.push(newMessage);
             run.messagesCount++;
         } else {
-            if (compareStrings(response.content, "")) {
+            if (!compareStrings(response.functionName, "")) {
+                toolRunning[runId] = response.functionName;
                 IOracle(oracleAddress).createFunctionCall(runId, response.functionName, response.functionArguments);
             } else {
-                Message memory newMessage;
-                newMessage.role = "assistant";
-                newMessage.content = response.content;
-                run.messages.push(newMessage);
-                run.messagesCount++;
+                toolRunning[runId] = "";
             }
+            Message memory newMessage;
```

Comment:
Dont we only want this when content is not "", atm we are doing this even if content is empty?